File: nihonapp2_11/lib/model/ClassRoom.dart
class Classroom {
  final int id;
  final String name;
  final String level;
  final String description;
  final int slMax;
  final String linkGiaotrinh;
  final DateTime start;
  final DateTime end;
  final int price;
  final String time;
  final String teacher;

  Classroom({
    required this.id,
    required this.name,
    required this.level,
    required this.description,
    required this.slMax,
    required this.linkGiaotrinh,
    required this.start,
    required this.end,
    required this.price,
    required this.time,
    required this.teacher,
  });

  factory Classroom.fromJson(Map<String, dynamic> json) {
    return Classroom(
      id: json['id'] ?? 0,
      name: json['name'] ?? 'Không có tên',
      level: json['level']['name'] ?? 'Không có cấp độ',
      description: json['description'] ?? 'Không có mô tả',
      slMax: (json['sl_max'] as num?)?.toInt() ?? 0,
      linkGiaotrinh: json['link_giaotrinh'] ?? '',
      start: json['start'] != null
          ? DateTime.parse(json['start'])
          : DateTime.now(),
      end: json['end'] != null ? DateTime.parse(json['end']) : DateTime.now(),
      price: (json['price'] as num?)?.toInt() ?? 0,
      time: json['time']['time'] ?? 'Không có thời gian',
      teacher: json['teacher']['teacherName'] ?? 'Không có giáo viên',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'level': {'name': level},
      'description': description,
      'sl_max': slMax,
      'link_giaotrinh': linkGiaotrinh,
      'start': start.toIso8601String(),
      'end': end.toIso8601String(),
      'price': price,
      'time': {'time': time},
      'teacher': {'teacherName': teacher},
    };
  }
}

File: app_tieng_nhat/src/main/java/com/example/app_tieng_nhat/model/ClassRoom.java
package com.example.app_tieng_nhat.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.Date;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@RequiredArgsConstructor
@Entity
@Table(name = "class")
public class ClassRoom {
    @Id
    private Long id;
    private String name;
    private String description;
    private Long sl_max;
    private String link_giaotrinh;
    private LocalDate start;
    private LocalDate end;
    private Long price;

    @ManyToOne
    @JsonManagedReference("level-class")
    @JoinColumn(name = "level_id")
    private Levels level;

    @ManyToOne
    @JsonManagedReference("class-time")
    @JoinColumn(name = "time_id",nullable = false)
    private Time time;

    @ManyToOne
    @JsonManagedReference("teacher-class")
    @JoinColumn(name = "teacher_id", nullable = false)
    private Teacher teacher;

    @ManyToMany
    @JoinTable(
            name = "user_class",
            joinColumns = @JoinColumn(name = "class_id"),
            inverseJoinColumns = @JoinColumn(name = "user_id")
    )
    private Set<Users> students = new HashSet<>();

    @OneToMany(mappedBy = "classRoom", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JsonBackReference("student-studentRegistration")
    @JsonIgnore
    private Set<StudentRegistration> studentRegistrations;

    @OneToMany(mappedBy = "classroom")
    @JsonManagedReference("home_work-class")
    private List<HomeWork> homeWorks;

}
